diff --git a/lib/index.js b/lib/index.js
index 4b0cfbe..3b0b8e7 100644
--- a/lib/index.js
+++ b/lib/index.js
@@ -488,6 +488,19 @@ function processBasic({ name, title, label, description, subSchema, rootSchema }
         field.field_type = 'storyblok-icon-sprite-picker-field-plugin';
         field.options = [{ name: 'allowedIcons', value: JSON.stringify(subSchema.enum || []), _uid: uuidv4() }];
     }
+    if (subSchema.type === 'string' && subSchema.format && subSchema.format === 'theme') {
+        field.type = 'custom';
+        field.field_type = 'storyblok-theme-select-field-plugin';
+        field.options = [];
+    }
+    if (subSchema.type === 'string' && subSchema.format && subSchema.format === 'sharepoint-folder') {
+        field.type = 'custom';
+        field.field_type = 'storyblok-sharepoint-folder-picker-field-plugin';
+        field.options = [];
+    }
+    if (subSchema.type === 'string' && subSchema.format && subSchema.format === 'textarea') {
+        field.type = 'textarea';
+    }
     if (subSchema.type === 'string' && subSchema.format && subSchema.format === 'image') {
         field.type = 'asset';
         field.filetypes = ['images'];
